src/util.rs

```diff
@@ -16,8 +22,69 @@ pub(crate) fn load_rustdoc_from_file(path: &Path) -> anyhow::Result<Crate> {
         .read_to_string(&mut s)
         .with_context(|| format!("Failed to read rustdoc JSON output file {:?}", path))?;
 
-    serde_json::from_str(&s)
-        .with_context(|| format!("Failed to parse rustdoc JSON output file {:?}", path))
+    match serde_json::from_str(&s) {
+        Ok(value) => Ok(value),
+        Err(_) => {
+            // Attempt to figure out the more precise reason the deserialization failed.
+            // Several possible options and their resolutions:
+            // (1) The file isn't actually a rustdoc JSON file. The user should supply a valid file.
+            // (2) The rustdoc JSON file has a version number that is too old, and isn't supported.
+            //     The user should upgrade to a newer nightly Rust version and regenerate the file.
+            // (3) The rustdoc JSON file has a version number that is too new, and isn't supported.
+            //     The user should attempt to upgrade to a newer cargo-semver-checks version
+            //     if one is already available, or open a GitHub issue otherwise.
+
+            // The error on this line is case (1).
+            let version = serde_json::from_str::<RustdocFormatVersion>(&s).with_context(|| {
```

Comment:
> @epage would you say this is also sufficient for https://github.com/obi1kenobi/cargo-semver-check/issues/88 or did you have something else in mind for that one? I wasn't sure if information about which nightly ranges use which rustdoc format versions is available somewhere, for example.

The core of #88 is the install instructions.

Something to keep in mind is some users might be stuck on a specific nightly version
- Ideally, we'd let them know what range of versions work rather than just saying "use latest"
- Ideally, they'd have a path for finding what older version would work

While the error reporting is good and important, the reason the documentation is important is that an error is a road bump and the more road bumps someone hits to accomplishing their task, the more frustrated they get.  #88 is about avoiding that.